serving_utils/client.py

```diff
@@ -44,50 +48,58 @@ def __init__(
         """Constructor.
 
         Args:
-            addr (str) : address of your serving
+            host (str) : hostname of your serving
+            port (int) : port of your serving
             pem: credentials of grpc
             channel+options: An optional list of key-value pairs (channel args in gRPC runtime)
             loop: asyncio event loop
             executor: a thread pool, or None to use the default pool of the loop
             standalone_pool_for_streaming: create a new thread pool (with 1 thread)
                 for each streaming method
         """
-        self.addr = addr
+        self.addr = f"{host}:{port}"
         if channel_options is None:
             channel_options = {}
         if pem is None:
-            self._channel = grpc.insecure_channel(
-                addr,
-                options=channel_options,
-            )
+            make_sync_channel = grpc.insecure_channel
         else:
             creds = grpc.ssl_channel_credentials(pem)
-            self._channel = grpc.secure_channel(
-                addr,
-                credentials=creds,
-                options=channel_options,
-            )
+            make_sync_channel = partial(grpc.secure_channel, credentials=creds)
 
         if loop is None:
             loop = asyncio.get_event_loop()
 
-        split_addr = addr.split(':')
-        host = split_addr[0]
-        port = split_addr[1]
-        # TODO: better addr parsing, secure channel
-        self._async_channel = Channel(host, port, loop=loop)
+        _, _, addrlist = socket.gethostbyname_ex(host)
```

Comment:
docstring:
Get all ips under the given host. One independent pyserving server will have one ip.
The number of addrList indicates the number of independent pyserving servers.